src/objects/HiddenText.js

```diff
@@ -1,17 +1,49 @@
 import Phaser from 'phaser';
-import Util from '../util';
 
 export default class HiddenText extends Phaser.GameObjects.Layer {
-    constructor(scene) {
+    constructor(scene, lightSource, radius) {
         super(scene);
         this.scene = scene;
+        this.lightSource = lightSource;
+        this.castRadius = radius;
 
-        const sceneCenter = Util.sceneCenter(scene);
+        const text1 = this.scene.add.image(495, 350, 'hidden-text');
 
-        const hiddenText = this.scene.add.image(sceneCenter.x, sceneCenter.y, 'hidden-text');
+        this.hiddenText = [ text1 ];
 
-        this.add([ hiddenText ]);
+        this.graphics = this.scene.make.graphics();
+        this.graphics.fillStyle(0xffffff);
+        // For some reason directly setting the starting location
+        // of the mask in the fillCircle() method call causes
+        // the circle to be offset from light source, so
+        // you have to use setPosition directly after.
+        // This is super annoying, especially as they both report
+        // having the same (x, y) co-ordinates but
+        // I don't have time to work out why right now.
+        this.graphics.fillCircle(0, 0, this.castRadius);
+        this.graphics.setPosition(this.lightSource.x, this.lightSource.y);
```

Comment:
Oh yeah these kinds of issues always crop up when we're messing about with etra graphics objects of containers, they've got their own x,y coordinate system inside themselves hich their children use, but they have their own x and y values which positions them in the scene itself.

I think you can just set the location when creating the graphics object, and the circle will always be at 0,0 inside the graphics object.

```suggestion
        this.graphics = this.scene.make.graphics({
            x: this.lightSource.x,
            y: this.lightSource.y
        });
        this.graphics.fillStyle(0xffffff);
        this.graphics.fillCircle(0, 0, this.castRadius);
```